src/language/parser.js

```diff
@@ -1404,6 +1417,22 @@ class Parser {
     }
   }
 
+  // Add the comment token to Parser
+  addCommentToParser() {
```

Comment:
Please inline it inside `advance` and also explain why you have nested `while (token.kind === TokenKind.COMMENT) {`

Reply:
Sorry! Nested `while` were by mistake, will remove it!